src/gluonnlp/data/transforms.py

```diff
@@ -693,3 +696,299 @@ def __call__(self, sample):
+                start_new_word = True
+            else:
+                if start_new_word:
+                    output.append([])
+                start_new_word = False
+                output[-1].append(char)
+            i += 1
+
+        return [''.join(x) for x in output]
+
+    def _is_punctuation(self, char):
+        """Checks whether `chars` is a punctuation character."""
+        cp = ord(char)
+        # We treat all non-letter/number ASCII as punctuation.
+        # Characters such as "^", "$", and "`" are not in the Unicode
+        # Punctuation class but we treat them as punctuation anyways, for
+        # consistency.
+        group0 = cp >= 33 and cp <= 47
+        group1 = cp >= 58 and cp <= 64
+        group2 = cp >= 91 and cp <= 96
+        group3 = cp >= 123 and cp <= 126
+        if (group0 or group1 or group2 or group3):
+            return True
+        cat = unicodedata.category(char)
+        if cat.startswith('P'):
+            return True
+        return False
+
+    def _is_whitespace(self, char):
+        """Checks whether `chars` is a whitespace character."""
+        # \t, \n, and \r are technically contorl characters but we treat them
+        # as whitespace since they are generally considered as such.
+        if char in [' ', '\t', '\n', '\r']:
+            return True
+        cat = unicodedata.category(char)
+        if cat == 'Zs':
+            return True
+        return False
+
+
+class BERTTokenizer(object):
+    r"""End-to-end tokenziation for BERT models.
+
+    Parameters
+    ----------
+    vocab : gluonnlp.Vocab or None, default None
+        Vocabulary for the corpus.
+    do_lower_case : bool, default True
+        If model is uncased, then do_lower_case should be set to True
+    max_input_chars_per_word : int, default 200
+
+    Examples
+    --------
+    >>> _,vocab = gluonnlp.model.bert_12_768_12(dataset_name='wiki_multilingual',pretrained=False)
+    >>> tokenizer = gluonnlp.data.BERTTokenizer(vocab=vocab)
+    >>> tokenizer(u"gluonnlp: 使NLP变得简单。")
+    ['gl', '##uo', '##nn', '##lp', ':', '使', 'nl', '##p', '变', '得', '简', '单', '。']
+
+    """
+
+    def __init__(self, vocab, do_lower_case=True, max_input_chars_per_word=200):
+        self.vocab = vocab
+        self.max_input_chars_per_word = max_input_chars_per_word
+        self.basic_tokenizer = BasicTokenizer(do_lower_case=do_lower_case)
+
+    def __call__(self, sample):
+        """
+
+        Parameters
+        ----------
+        sample: list(str)
+            The sentence to detokenize
+
+        Returns
+        -------
+        ret : str
+            Detokenized text
+        """
+        return self._tokenizer(sample)
+
+    def _tokenizer(self, text):
+        split_tokens = []
+        for token in self.basic_tokenizer(text):
+            for sub_token in self._WordpieceTokenizer(token):
+                split_tokens.append(sub_token)
+
+        return split_tokens
+
+    def _WordpieceTokenizer(self, text):
```

Comment:
suggest either make it class `_WordpieceTokenizer` with `__call__` method, or change it to `def _tokenize_wordpiece`

Reply:
If making into a separate, public class, the document format needs to be fixed. GluonNLP adopts the numpy style docstring.